### PR TITLE
(chart-api) fix for fetching sector benchmarks + clean up

### PR DESCRIPTION
**Pull request by kowal**
Extracted query helpers:
- `Sector#latest_released_benchmarks`
- `Company#latest_sector_benchmarks`
- `Company#latest_sector_benchmarks_before_last_assessment`

Also, updated docs.

### CODE REVIEW

File: app/models/company.rb [hunk 1]
```diff
@@ -55,7 +55,37 @@ def latest_cp_assessment
...
+  #
+  # @example Company has assessment:
+  # - benchmarks available for 04.2017 and 05.2018
+  # - if assessment date is 06.2018 - we take benchmarks from 05.2018
+  # - if assessment date is 06.2017 - we take benchmarks from 04.2017
+  def latest_sector_benchmarks_before_last_assessment
+    last_assessment_date = latest_cp_assessment&.assessment_date
+    sector_benchmarks_dates = sector.cp_benchmarks.pluck(:release_date).uniq.sort
+
+    if last_assessment_date
```
Comment (agwozdowski) [inline app/models/company.rb]: I prefer to use early return instead big if, like:
```
unless last_assessment_date
  return 

last_release_date_before_assessment =
....
```

File: app/models/sector.rb [hunk 2]
```diff
@@ -19,4 +19,8 @@ class Sector < ApplicationRecord
 
   validates_presence_of :name, :slug
   validates_uniqueness_of :name, :slug
+
+  def latest_released_benchmarks
```
Comment (tsubik) [inline app/models/sector.rb]: :+1:

File: spec/models/company_spec.rb [hunk 3]
```diff
@@ -86,4 +86,63 @@
       expect(company.latest_mq_assessment.assessment_date.to_s).to eq('2019-05-01')
     end
   end
+
+  describe '#latest_sector_benchmarks_before_last_assessment' do
```
Comment (tsubik) [inline spec/models/company_spec.rb]: :+1:

File: app/models/company.rb [hunk 4]
```diff
@@ -55,7 +55,36 @@ def latest_cp_assessment
...
-  def sector_benchmarks
-    sector.cp_benchmarks
+  def latest_sector_benchmarks
+    sector.latest_released_benchmarks
+  end
+
+  # Returns sector CP benchmarks:
+  # - if company has assessment      - from the last date before latest CP::Assessment date
+  # - if company has no assessments  - from latest benchmarks
+  # - if company has old assessments - from latest benchmarks
```
Comment (tsubik) [inline app/models/company.rb]: what does it mean "old assessments" here?

Comment (kowal) [reply app/models/company.rb]: old = "from before the oldest benchmark", I'll update it.

Comment (tsubik) [reply app/models/company.rb]: so then the oldest benchmark should be used, right?

Comment (tsubik) [reply app/models/company.rb]: ahh, no, then I don't know what should be used :)

**Comment by kowal** [comment]
waiting from @simaob confirmation if that's what we want here :)

**Comment by simaob** [comment]
The updated comment looks good to me and makes sense! Thanks for sorting it out.

**Pull request merged**